pkg/meta/tkv.go

```diff
@@ -3756,3 +3758,146 @@ func (m *kvMeta) loadDumpedACLs(ctx Context) error {
 		return nil
 	})
 }
+
+type kvEntryScanner struct {
+	cache       []*Entry
+	cnt         int
+	m           *kvMeta
+	defaultCap  int
+	parentInode Ino
+	plus        bool
+	bgChan      chan []*Entry
+}
+
+func (m *kvMeta) NewMetaEntryScanner(inode Ino, plus bool) EntryScanner {
+	c := &kvEntryScanner{
+		cache:       make([]*Entry, 0),
+		cnt:         0,
+		m:           m,
+		parentInode: inode,
+		defaultCap:  DefaultCap,
+		plus:        plus,
+		bgChan:      nil,
+	}
+	return c
+}
+
+func (sc *kvEntryScanner) Valid() bool {
+	return true
+}
+
+func (sc *kvEntryScanner) Close() {
+	if sc.bgChan != nil {
+		_ = <-sc.bgChan
+		sc.cache = make([]*Entry, 0)
+		sc.cnt = 0
+		sc.bgChan = nil
+	}
+}
+
+func (sc *kvEntryScanner) scanEntries(startKey, endKey []byte, limit int) ([]*Entry, error) {
+	if limit <= 0 {
+		return nil, nil
+	}
+	entries := make([]*Entry, 0)
+	prefix := len(sc.m.entryKey(sc.parentInode, ""))
+	err := sc.m.client.txn(func(tx *kvTxn) error {
+		var cnt int = 0
+		tx.scan(startKey, endKey, false, func(name, buf []byte) bool {
+
+			typ, ino := sc.m.parseEntry(buf)
+			if len(name) == prefix {
+				logger.Errorf("Corrupt entry with empty name: inode %d parent %d", ino, sc.parentInode)
+				return false
+			}
+			entries = append(entries, &Entry{
+				Inode: ino,
+				Name:  []byte(name)[prefix:],
+				Attr:  &Attr{Typ: typ},
+			})
+			cnt++
+			return cnt < limit
+		})
+		return nil
+	}, 0)
+	return entries, err
+}
+
+func (sc *kvEntryScanner) fillAttr(entries []*Entry) (err error) {
```

Comment:
change and reuse fillAttr

Reply:
`fillAttr` is a lambda function https://github.com/juicedata/juicefs/blob/main/pkg/meta/tkv.go#L1799  , it is ok to extract it as a new function `func (m *kvMeta) fillAttres([]*Entry) error `

Reply:
yes